jwst/wfss_contam/observations.py

```diff
@@ -0,0 +1,545 @@
+            pixels with no source signal and an integer source number for pixels with signal
+        config: Full path name of a NIRCam GRISMCONF configuration file
+        mod: NIRCam Module: A or B
+        order: The spectral order to simulate, +1 or +2 for NIRCAM
+        max_split: Number of chunks to compute instead of trying everything at once.
+        SED_file: Name of HDF5 file containing datasets matching the ID in the segmentation file
+                  and each consisting of a [[lambda],[flux]] array.
+        extrapolate_SED:
+        max_cpu:
+        ID:
+        SBE_save: If set to a path, HDF5 containing simulated stamps for all obsjects will be saved.
+        boundaries: a tuple containing the coordinates of the FOV within the larger seed image.
+        renormalize:
+        """
+
+        # This loads all the info from the configuration file for this grism mode;
+        # things like the name of the POM mask file, the dispersion coeffs, etc.
+        self.seg_wcs = segmap_model.meta.wcs
+        self.grism_wcs = grism_wcs
+        self.ID = ID
+        self.IDs = []
+        self.dir_image_names = direct_images
+        self.seg = segmap_model.data
+        # self.dims = np.shape(self.seg)
+        self.dims = [2048, 2048]
+        self.filter = filter
+        self.order = order
+        self.SED_file = SED_file   # should always be NONE for baseline pipeline (use flat SED)
+        self.SBE_save = SBE_save   # may want to have a new product in which simulated spectra are saved (NOT HDF5 format!)
+        self.max_cpu = max_cpu
+        self.cache = False
+        self.POM_mask = None
+        self.POM_mask01 = None
+        self.renormalize = renormalize
+
+        # Get the wavelength range for the grism image
+        wavelength_range = wr_ref.get_wfss_wavelength_range(self.filter, [self.order])
+        self.wmin = wavelength_range[self.order][0]
+        self.wmax = wavelength_range[self.order][1]
+        print("wmin, wmax:", self.wmin, self.wmax)
+
+        if len(boundaries) != 4:
+            # self.NAXIS = [2048, 2048]
+            # xpad = (np.shape(self.seg)[1] - self.NAXIS[0])//2
+            # ypad = (np.shape(self.seg)[0] - self.NAXIS[1])//2
+            # self.xstart = 0 + xpad
+            # self.xend = xpad + self.NAXIS[0] - 1
+            # self.ystart = 0 + ypad
+            # self.yend = ypad + self.NAXIS[1] - 1
+            # print(f"No boundaries passed. Assuming symmetrical padding of {xpad} {ypad} pixels")
+            # print(f"and a final size of {self.xend+1-self.xstart} {self.yend+1-self.ystart}.")
+            self.xstart = 0
+            self.xend = self.xstart + self.dims[0] - 1
+            self.ystart = 0
+            self.yend = self.ystart + self.dims[1] - 1
+            print("No boundaries passed.")
+            print(f"Using final size of {self.xend+1-self.xstart} {self.yend+1-self.ystart}.")
+        else:
+            self.xstart, self.xend, self.ystart, self.yend = boundaries
```

Comment:
Is `boundaries` equivalent to the `footprint`?

Reply:
No (at least I don't think so). It's a simple list like [0, 2047, 0, 2047], which is used to tell the routine what size to use to create the grism image that will contain the simulated spectra. Can't just use the size of the direct image (or equivalently the segmentation map), because those will often be larger due to mosaicing/dithering. But we need/want the equivalent of a single full-frame grism image. And can't get it from the input grism image that's being corrected, because it's already been chopped up into little pieces by `extract_2d`.